docs/tutorials/typescript/subscriptions/index.md

```diff
@@ -1,416 +1,417 @@
+Add the following code to the `shared.ts` file to define the object, as well as the Task Queue name:
 
+<!--SNIPSTART subscription-ts-shared-file {"selectedLines": ["1-14"]}-->
+[src/shared.ts](https://github.com/temporalio/subscription-workflow-project-template-typescript/blob/main/src/shared.ts)
 ```ts
-// /src/workflows.ts
-import { sleep, proxyActivities } from '@temporalio/workflow';
-import type * as activities from './activities';
-
-const { sendWelcomeEmail, sendSubscriptionOverEmail } = proxyActivities<
-  typeof activities
->({
-  startToCloseTimeout: '1 minute',
-});
+export const TASK_QUEUE_NAME = "subscriptions-task-queue";
 
-export async function SubscriptionWorkflow(
-  email: string,
-  trialPeriod: string | number
-) {
-  await sendWelcomeEmail(email);
-  await sleep(trialPeriod);
-  await sendSubscriptionOverEmail(email);
+export interface Customer {
+  firstName: string;
+  lastName: string;
+  email: string;
+  subscription: {
+    trialPeriod: number;
+    billingPeriod: number;
+    maxBillingPeriods: number;
+    initialBillingPeriodCharge: number;
+  };
+  id: string;
 }
 ```
+<!--SNIPEND-->
 
-Here we are using the `sleep` API for the first time.
-It does what it says; defers execution for a preset time (note that it accepts both a string or number of milliseconds, [per its docs](https://docs.temporal.io/typescript/workflows#sleep)).
+Now that you have your `Customer` object defined, you can move on to writing the Workflow Definition. The Workflow will orchestrate the sequence of Activities such as sending welcome emails, charging customers, and handling cancellations.
 
-To test this out, you will also have to modify your Client code accordingly:
+To create a new Workflow Definition, create a new file called `workflows.ts`. This file will contain the `subscriptionWorkflow` function. The requirements of this Workflow Definition are the following:
 
+1. **User Signup and Free Trial**: When the user signs up, send a welcome email and start a free trial for the duration defined by trialPeriod.
+2. **Cancellation During Trial**: If the user cancels during the trial period, send a trial cancellation email and complete the Workflow.
+3. **Billing Process**:
+   - If the trial period expires without cancellation, start the billing process.
+   - As long as the number of billing periods does not exceed  `maxBillingPeriods`:
+    - Charge the customer the amount defined by `billingPeriodChargeAmount`.
+    - Wait for the next `billingPeriod`.
+   - If the customer cancels during a billing period, **send a subscription cancellation email**.
+   - If Subscription has ended normally (exceeded `maxBillingPeriods` without cancellation), send a subscription ended email and complete the Workflow.
+   - If the subscription ends normally (exceeding `maxBillingPeriods` without cancellation), send a subscription ended email and complete the Workflow.
+4. **Dynamic Updates**: At any point while subscriptions are ongoing, be able to:
```

Comment:
These bullet points look a little off in the preview for #4. I don't think they're aligned with #3.